### PR TITLE
Add a script to run mpt with FasterTransformer

### PR DESCRIPTION
**Pull request by dskhudia**
Adds a script to run a MPT checkpoint in FT format. 

Test to be added (Basically convert a 125m param model to FT and run it and compare outputs) in a future PR. Model run passes vibe check on some sample inputs. 

Fixes https://github.com/mosaicml/llm-foundry/issues/250 and https://github.com/mosaicml/llm-foundry/issues/67

### CODE REVIEW

File: scripts/inference/convert_hf_mpt_to_ft.py [hunk 1]
```diff
@@ -200,37 +205,46 @@ def convert_mpt_to_ft(model_name_or_path: str, output_dir: str,
...
+                'qk_ln is enabled for this MPT model. This may not work as expected in FT. Use --force to force a conversion.'
+            )
+
+        with open(os.path.join(save_dir, 'config.ini'), 'w') as configfile:
             config.write(configfile)
     except:
         print(f'Failed to save the config in config.ini.')
         raise
 
+    sys.exit(1)
```
Comment (abhi-mosaic) [inline scripts/inference/convert_hf_mpt_to_ft.py:247]: Is this intended to be here?

File: scripts/inference/run_mpt_with_ft.py [hunk 2]
```diff
@@ -0,0 +1,476 @@
+# Copyright 2022 MosaicML LLM Foundry authors
```
Comment (abhi-mosaic) [inline scripts/inference/run_mpt_with_ft.py:1]: Naming question -- does this script support any FT model or only MPT? If the former I might suggest renaming this script to something like `ft_generate.py` to signal that this script is analogous to `hf_generate.py`

**Review by abhi-mosaic — COMMENTED** [review]
Have not tested myself yet so will refrain from approving just yet. Left small comments

File: scripts/inference/run_mpt_with_ft.py (hunk 2 above)
Comment (dskhudia) [reply scripts/inference/run_mpt_with_ft.py:1]: This only supports MPT model.

File: scripts/inference/convert_hf_mpt_to_ft.py (hunk 1 above)
Comment (dskhudia) [reply scripts/inference/convert_hf_mpt_to_ft.py:247]: Good catch. No. It was for some local testing I was doing. Removing it.

**Review by abhi-mosaic — APPROVED** [review]

**Comment by TriLoo** [comment]
I have checked this script, can get same outputs from FasterTransformer as pytorch.

**Comment by dskhudia** [comment]
@TriLoo : Wonderful. Thanks for checking. Do you mind sharing your setup/script for comparing?

**Comment by TriLoo** [comment]
> @TriLoo : Wonderful. Thanks for checking. Do you mind sharing your setup/script for comparing?

sorry, I cannot share the script, but the official bloom example from FasterTransformer can be a good reference.

**Pull request merged**